File: fabric/src/main/kotlin/cache/CallData.kt
package com.obecto.perper.fabric.cache

class CallData(
    var agent: String,
    var instance: String,
    var delegate: String,
    // var parameters: Array<Any?>,
    var callerAgent: String,
    var caller: String,
    var localToData: Boolean,
    var finished: Boolean,
    // var result: Array<Any?>?,
    var error: String?,
)

File: functions/runtime/src/Perper.WebJobs.Extensions/Cache/CallData.cs
namespace Perper.WebJobs.Extensions.Cache
{
    public class CallData : IInstanceData
    {
        public string Agent { get; set; }
        public string AgentDelegate { get; set; }
        public string Delegate { get; set; }
        public string CallerAgentDelegate { get; set; }
        public string Caller { get; set; }
        public bool Finished { get; set; }
        public bool LocalToData { get; set; }
        public object?[]? Parameters { get; set; }
        public object? Result { get; set; }
        public string? Error { get; set; }
    }
}